src/areas/Login/LoginModal.js

```diff
@@ -0,0 +1,45 @@
+import React, {useState} from 'react';
+import {Backdrop, Box, Modal, Fade, Button} from '@mui/material';
+import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
+import { uiConfig, auth } from './FirebaseConfig';
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+};
+
+export default function LoginModal({isSignedIn, setIsSignedIn}) {
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  return (
+    <div>
+      <Button onClick={handleOpen}>Login</Button>
+      <Modal
```

Comment:
Why is this a `Modal` rather than a `Dialog`?
https://mui.com/material-ui/react-dialog/#basic-dialog